browser/DaisyPayments.js

```diff
@@ -47,41 +109,94 @@ export default class DaisyPayments extends ClientPayments {
+ * const daisy = await DaisySDK.initPayments({
+ *   manager: { identifier: ... }, withGlobals: { web3 },
+ * });
+ *
+ * console.log(daisy.prepareToken(daisy.loadToken(invoice)) instanceof DaisyPaymentsOnToken);
+ * // > true
+ */
+export class DaisyPaymentsOnToken extends DaisyPayments {
+  /**
+   * @private
+   */
+  constructor({ manager, token, override, withGlobals }) {
+    super({ manager, override, withGlobals });
     this.token = token;
   }
 
-  balanceOf(sendArgs) {
-    if (!sendArgs || !sendArgs.tokenOwner) {
-      throw new TypeError(`balanceOf() was called without a tokenOwner specified. Be sure to call balanceOf() like:
-      
-      daisy
-        .prepareToken(token)
-        .balanceOf({ tokenOwner: account })
-      
-      `);
-    }
-    return this.token.methods["balanceOf"](sendArgs.tokenOwner).call();
-  }
-
   /**
    * Transfer token to an address "predicted" by create2.
    * @async
+   * @param {module:common~PaymentInvoice|Object} invoice - Input object
+   * @param {string|number} invoice.invoicedPrice - Amount to pay/transfer. See: {@link module:common~PaymentInvoice#invoicedPrice}.
+   * @param {string} invoice.address - Beneficiary of the transfer. See: {@link module:common~PaymentInvoice#address}.
+   * @param {Object} sendArgs - Web3 arguments for transactions. Must have `from` field. @see {@link https://web3js.readthedocs.io/en/1.0/web3-eth-contract.html#methods-mymethod-send|web3js.readthedocs}
+   * @param {string} sendArgs.from - User account Ethereum address (payer).
+   * @returns {external:PromiEvent} It's a transaction result object.
+   *
+   * @example
+   *
+   * const daisy = await DaisySDK.initPayments({
+   *   manager: { identifier: ... }, withGlobals: { web3 },
+   * });
+   * const token = daisy.loadToken(invoice);
+   * const { transactionHash } = await daisy
+   *   .prepareToken(token)
+   *   .pay(invoice, { from: account });
    */
   pay(invoice, sendArgs) {
+    if (!invoice) {
+      throw new TypeError("Missing `invoice` argument.");
+    } else if (!sendArgs || !sendArgs.from) {
+      throw new TypeError("Missing `sendArgs.from` argument");
+    }
+
     const price = invoice["invoicedPrice"];
     const address = invoice["address"];
     return this.token.methods["transfer"](address, price).send(sendArgs);
   }
 
+  /**
+   * Get blockchain's transfers. Useful to check the state of a processing transaction.
+   * @async
+   * @param {module:common~PaymentInvoice|Object} invoice - Input object
+   * @param {string} invoice.address - Beneficiary of the transfer. See: {@link module:common~PaymentInvoice#address}.
+   * @param {Object} [opts={}]
+   * @param {number|string} [opts.fromBlock=0]
+   * @param {string} [opts.toBlock="latest"]
+   * @returns {external:PromiEvent} Web3 events as an array.
+   */
   getTransfers(invoice, opts = { fromBlock: 0, toBlock: "latest" }) {
```

Comment:
Jw, why do we call this a transfer and not a payment? This is for getting all the payments to an invoice during a given time period, no?

Reply:
oh "transfers" are "processing actions" on the blockchain that will become a Receipt after some confirmations